### DISCUSSION TITLE
Understanding Data and Prisma

### DISCUSSION THREAD

**Discussion by heydavee** [remix-run/remix#6532] [Q&A] [answered]
Hey, i am currently trying to build an application with remix and i am somewhat stuck trying to understand data and how to access it

I have a model student which looks like this:

```
model Student {
  id             Int              @id @default(autoincrement())
  studentId      Int
  firstName      String
  lastName       String
  birthDate      DateTime
  birthPlace     String
  nationality    String
  street         String
  zip            String
  city           String
  contact        Contact[]
}
```

My Contact model looks like this:

```
model Contact {
  id        Int     @id @default(autoincrement())
  phone     String?
  mobile    String?
  email     String?
  studentId Int
  Student   Student @relation(fields: [studentId], references: [id])
}
```

Now when i am trying to access the data via a loader, contact is missing in my student data

```
export const loader = async () => {
  return json({
    students: await db.student.findMany(),
  });
};
```

When i am using select, it works but my types, which i generated with `prisma generate`, are missing contact. I am so confused.

```
export type Student = {
  id: number
  studentId: number
  firstName: string
  lastName: string
  birthDate: Date
  birthPlace: string
  nationality: string
  street: string
  zip: string
  city: string
}
```

What am i doing / understanding wrong?

**Comment by machour** [comment, marked as answer]
This seems like a usage problem with Prisma, specifically with relations. 
Give https://www.prisma.io/docs/concepts/components/prisma-schema/relations a read, and ask support on Prisma's repo
